pkg/validation/cluster/handler.go

```diff
@@ -68,11 +73,18 @@ func (h *AdmissionHandler) Handle(_ context.Context, req webhook.AdmissionReques
 		if err != nil {
 			return admission.Errored(http.StatusBadRequest, err)
 		}
+
 		validationErr := h.validateCreateOrUpdate(cluster)
 		if validationErr != nil {
 			h.log.Info("cluster admission failed", "error", validationErr)
 			return webhook.Denied(fmt.Sprintf("cluster validation request %s rejected: %v", req.UID, validationErr))
 		}
+
+		if err := h.rejectEnableUserSSHKeyAgentUpdate(ctx, cluster); err != nil {
+			h.log.Info("cluster admission failed", "error", err)
+			return webhook.Denied(fmt.Sprintf("cluster validation request %s rejected: %v", req.UID, err))
+		}
```

Comment:
Why is this a dedicated function and not part of the existing `validateCreateOrUpdate`?